### PR TITLE
Preview module multi

### PR DESCRIPTION
**Pull request by kuldev**

### CODE REVIEW

File: modules/ezcontent_preview/ezcontent_preview.module [hunk 1]
```diff
@@ -50,3 +51,20 @@ function ezcontent_preview_menu_local_tasks_alter(&$data, $route_name) {
         }
     }
 }
+*/
```
Comment (vaibhavjain-in) [inline modules/ezcontent_preview/ezcontent_preview.module]: Can we remove the commented code ?

File: modules/ezcontent_preview/ezcontent_preview.module [hunk 2]
```diff
@@ -50,3 +51,20 @@ function ezcontent_preview_menu_local_tasks_alter(&$data, $route_name) {
...
+    $label = 'Published or Access Unpublished Token';
+    $data['node']['published_access_unpublished_token'] = array(
+        'title' => $label,
+        'filter' => array(
+            'title' => $label,
+            'help' => t('Published or Access unpublish content with token'),
+            'field' => 'nid',
+            'id' => 'published_access_unpublished_token'
+        ),
+    );
```
Comment (vaibhavjain-in) [inline modules/ezcontent_preview/ezcontent_preview.module]: Lets format it to 2 spacing.

File: modules/ezcontent_preview/src/Controller/PreviewView.php [hunk 3]
```diff
@@ -5,30 +5,43 @@
...
-          'library' => [
-            'ezcontent_preview/global',
-          ],
-        ],
-      );
-    }
+  public function preview(NodeInterface $node = NULL, $preview_type) {
+
+    if ($preview_type) {
+      $decoupledRoutes = \Drupal::entityTypeManager()->getStorage('ezcontent_preview')->load($preview_type);
```
Comment (vaibhavjain-in) [inline modules/ezcontent_preview/src/Controller/PreviewView.php]: Can we inject the entity type manager object.

File: modules/ezcontent_preview/src/Controller/PreviewView.php [hunk 4]
```diff
@@ -5,30 +5,43 @@
...
+            '#markup' => $output,
+            '#attached' => [
+              'library' => [
+                'ezcontent_preview/global',
+              ],
+            ],
+          );
+        }
+      }
+    }
```
Comment (vaibhavjain-in) [inline modules/ezcontent_preview/src/Controller/PreviewView.php]: Lets format it to 2 spacing.

File: modules/ezcontent_preview/src/Entity/Preview.php [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+   * @var string
+   */
+  public $token_time;
+
+  /**
+   *
+   * @var string
+   */
+  public $content_entity;
+}
```
Comment (vaibhavjain-in) [inline modules/ezcontent_preview/src/Entity/Preview.php]: Lets add a new line at EOF.

File: modules/ezcontent_preview/src/Form/PreviewDeleteForm.php [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+namespace Drupal\ezcontent_preview\Form;
+
+use Drupal\Core\Entity\EntityConfirmFormBase;
+use Drupal\Core\Url;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Builds the form to delete
+ */
+
```
Comment (vaibhavjain-in) [inline modules/ezcontent_preview/src/Form/PreviewDeleteForm.php:12]: This new line is not required. lets remove it.

File: modules/ezcontent_preview/src/Form/PreviewForm.php [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+    $form['token_time'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Token time'),
+      '#maxlength' => 10,
+      '#default_value' => $preview->token_time,
+      '#description' => $this->t("Time in seconds to keep the token alive."),
+      '#required' => TRUE,
+    ];
+
+    $nodeTypes = \Drupal\node\Entity\NodeType::loadMultiple();
```
Comment (vaibhavjain-in) [inline modules/ezcontent_preview/src/Form/PreviewForm.php:68]: Lets inject this object

File: modules/ezcontent_preview/src/Form/PreviewForm.php [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+        '%label' => $preview->label(),
+      ]));
+    }
+    else {
+      $this->messenger()->addMessage($this->t('The %label was not saved.', [
+        '%label' => $preview->label(),
+      ]), MessengerInterface::TYPE_ERROR);
+    }
+
+
```
Comment (vaibhavjain-in) [inline modules/ezcontent_preview/src/Form/PreviewForm.php:120]: Lets remove multiple blank lines.

File: modules/ezcontent_preview/src/Plugin/Derivative/DynamicLocalTasks.php [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+/**
+ * Defines dynamic local tasks.
+ */
+class DynamicLocalTasks extends DeriverBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getDerivativeDefinitions($base_plugin_definition) {
+    $decoupledRoutes = \Drupal::entityTypeManager()->getStorage('ezcontent_preview')->loadMultiple();
```
Comment (vaibhavjain-in) [inline modules/ezcontent_preview/src/Plugin/Derivative/DynamicLocalTasks.php:18]: Lets inject this object

File: modules/ezcontent_preview/src/Plugin/Derivative/DynamicLocalTasks.php [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+    if ($decoupledRoutes) {
+      $current_path = \Drupal::service('path.current')->getPath();
```
Comment (vaibhavjain-in) [inline modules/ezcontent_preview/src/Plugin/Derivative/DynamicLocalTasks.php:21]: Lets inject this object

File: modules/ezcontent_preview/src/Plugin/Derivative/DynamicLocalTasks.php [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+      $path_args = explode('/', $current_path);
+      if ($path_args && $path_args[1] == 'node') {
+        $node = \Drupal\node\Entity\Node::load($path_args[2]);
```
Comment (vaibhavjain-in) [inline modules/ezcontent_preview/src/Plugin/Derivative/DynamicLocalTasks.php:24]: Lets inject this object

File: modules/ezcontent_preview/src/Plugin/Derivative/DynamicLocalTasks.php [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+              'weight' => 100,
+            ];
+          }
+        }
+      }
+    }
+    return parent::getDerivativeDefinitions($base_plugin_definition);
+  }
+
+}
```
Comment (vaibhavjain-in) [inline modules/ezcontent_preview/src/Plugin/Derivative/DynamicLocalTasks.php]: Lets add a new line at EOF

File: modules/ezcontent_preview/src/Utils.php [hunk 13]
```diff
@@ -13,9 +13,10 @@
  */
 class Utils {
 
-  public function buildUrl(NodeInterface $node = NULL, $options = []) {
+  public function buildUrl(NodeInterface $node = NULL, $decoupledRoutes, $options = []) {
     $config = \Drupal::config('ezcontent_preview.settings');
-    $preview_base_url = $config->get('ezcontent_preview_url');
+    //$preview_base_url = $config->get('ezcontent_preview_url');
```
Comment (vaibhavjain-in) [inline modules/ezcontent_preview/src/Utils.php]: Lets remove the commented code.

**Review by vaibhavjain-in — CHANGES_REQUESTED** [review]
@kuldev Lets fix some minor issues here.

File: modules/ezcontent_preview/ezcontent_preview.module (hunk 1 above)
Comment (kuldev) [reply modules/ezcontent_preview/ezcontent_preview.module]: fixed

File: modules/ezcontent_preview/ezcontent_preview.module (hunk 2 above)
Comment (kuldev) [reply modules/ezcontent_preview/ezcontent_preview.module]: fixed

File: modules/ezcontent_preview/src/Controller/PreviewView.php (hunk 3 above)
Comment (kuldev) [reply modules/ezcontent_preview/src/Controller/PreviewView.php]: added DI for it.

File: modules/ezcontent_preview/src/Controller/PreviewView.php (hunk 4 above)
Comment (kuldev) [reply modules/ezcontent_preview/src/Controller/PreviewView.php]: done

File: modules/ezcontent_preview/src/Entity/Preview.php (hunk 5 above)
Comment (kuldev) [reply modules/ezcontent_preview/src/Entity/Preview.php]: done

File: modules/ezcontent_preview/src/Plugin/Derivative/DynamicLocalTasks.php (hunk 9 above)
Comment (kuldev) [reply modules/ezcontent_preview/src/Plugin/Derivative/DynamicLocalTasks.php:18]: not possible I believe as on dynamic tabs not getting any DI feature.

File: modules/ezcontent_preview/src/Plugin/Derivative/DynamicLocalTasks.php (hunk 10 above)
Comment (kuldev) [reply modules/ezcontent_preview/src/Plugin/Derivative/DynamicLocalTasks.php:21]: not possible I believe as on dynamic tabs not getting any DI feature.

File: modules/ezcontent_preview/src/Plugin/Derivative/DynamicLocalTasks.php (hunk 11 above)
Comment (kuldev) [reply modules/ezcontent_preview/src/Plugin/Derivative/DynamicLocalTasks.php:24]: not possible I believe as on dynamic tabs not getting any DI feature.

File: modules/ezcontent_preview/src/Plugin/Derivative/DynamicLocalTasks.php (hunk 12 above)
Comment (kuldev) [reply modules/ezcontent_preview/src/Plugin/Derivative/DynamicLocalTasks.php]: done

File: modules/ezcontent_preview/src/Utils.php (hunk 13 above)
Comment (kuldev) [reply modules/ezcontent_preview/src/Utils.php]: done

**Review by vaibhavjain-in — APPROVED** [review]
Thanks for the changes @kuldev

**Pull request merged**